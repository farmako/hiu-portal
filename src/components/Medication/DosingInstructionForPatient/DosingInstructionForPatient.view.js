import React from 'react';
import * as PropTypes from 'prop-types';

const DosingInstructionForPatient = (props) => {
  const { dosage } = props;
  if (dosage.patientInstruction) {
    return (
      <li>
        <span>
          Instruction to Patient:
          {dosage.patientInstruction}
        </span>
      </li>
    );
  }
  return null;
};

DosingInstructionForPatient.propTypes = {
  dosage: PropTypes.shape({
    patientInstruction: PropTypes.string,
  }),
};

DosingInstructionForPatient.defaultProps = {
  dosage: {
    patientInstruction: '',
  },
};

export default DosingInstructionForPatient;
