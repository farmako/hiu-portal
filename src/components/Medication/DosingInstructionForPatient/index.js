import DosingInstructionForPatient from './DosingInstructionForPatient.view';

export default DosingInstructionForPatient;
