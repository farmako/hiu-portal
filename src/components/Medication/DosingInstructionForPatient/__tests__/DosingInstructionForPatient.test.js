import React from 'react';
import { shallow } from 'enzyme';
import DosingInstructionForPatient from '../DosingInstructionForPatient.view';

describe('DosingInstructionsForPatient', () => {
  const mockData = {
    patientInstruction: 'Information text ...',
  };
  const wrapper = shallow(
    <DosingInstructionForPatient dosage={mockData} />,
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render data of patient instruction', () => {
    expect(wrapper.find('span').at(0).text()).toEqual('Instruction to Patient:Information text ...');
  });

  it('should not render anything if patientInstruction is not provided', () => {
    mockData.patientInstruction = null;
    wrapper.setProps({ dosage: mockData });
    expect(wrapper.find('li').at(0).exists()).toEqual(false);
  });
});
