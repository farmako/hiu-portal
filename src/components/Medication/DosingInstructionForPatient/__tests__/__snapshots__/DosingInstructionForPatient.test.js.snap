// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DosingInstructionsForPatient should render properly 1`] = `
<li>
  <span>
    Instruction to Patient:
    Information text ...
  </span>
</li>
`;
